### DISCUSSION TITLE
Textual on Android (Termux)

### DISCUSSION THREAD

**Discussion by kilacoda** [Textualize/textual#583] [Show and tell]
Hi,

Didn't find anyone posting about it here, but Textual seems to work well (at least the `textual.app` module, see attached video) on Android via termux. Don't know if there's any practical usecase and/or demand for TUIs on smartphones, but hey it works!

https://user-images.githubusercontent.com/14952412/174221166-e97978a9-f996-4147-8a96-b8b8890ce180.mp4

**Comment by timothywcrane** [comment]
good to know. I can think of plenty of uses, starting with a front end for headless briar without JS. I wonder (thinking out loud will be sure to try) if XLST templating for punch outs would format well ootb.

**Comment by Aljoberg** [comment]
As somebody who uses Termux as a second computer (when I don't have mine) and IDE, this is really useful to me everyday. Lovin it!

**Comment by ipfilip** [comment]
Looks like it might become even easier with android terminal from google: https://www.androidauthority.com/android-linux-terminal-app-3489887/

**Comment by Disonantemus** [comment]
> Didn't find anyone posting about it here, but Textual seems to work well (at least the `textual.app` module, see attached video) on Android via termux. Don't know if there's any practical usecase and/or demand for TUIs on smartphones, but hey it works!

How did you installed it?

I tried and fail.
Seems related to tree-sitter (Termux has some packages, not all of them).

I did and [Issue in **Textual** repo](https://github.com/Textualize/textual/issues/5644) but I got zero feedback.

My use case is use **ParLlama**, a terminal LLM chat and manager like aichat/oterm (both work in Termux using ollama and maybe llama-cpp). Textual is a dependency.

**Comment by Yurii-huang** [comment]
so cool

**Comment by tomesparon** [comment]
clicking on URLs seems to not function for termux
